src/content/en/updates/2016/12/imagecapture.md

```diff
@@ -0,0 +1,349 @@
+
+    navigator.mediaDevices.getUserMedia({video: true})
+      .then(gotMedia)
+      .catch(error => console.error('getUserMedia() error:', error));
+
+    function gotMedia(mediaStream) {
+      const mediaStreamTrack = mediaStream.getVideoTracks()[0];
+      const imageCapture = new ImageCapture(mediaStreamTrack);
+      console.log(imageCapture);
+    }
+
+You can try out this code from the DevTools console.
+
+Note: To choose between different cameras, such as the front and back camera on
+a phone, get a list of available devices via the
+`MediaDevices.enumerateDevices()` method, then set `deviceId` in `getUserMedia()`
+constraints — as per the demo [here](https://webrtc.github.io/samples/src/content/devices/input-output/).
+
+You can use `takePhoto()` to get a still image and then set it as the `src` of
+an `<img>`:
+
+    imageCapture.takePhoto()
+      .then(blob => {
+        image.src = URL.createObjectURL(blob);
+      })
+      .catch(error => console.error('takePhoto() error:', error));
+
+Use `grabFrame()` to get data for a frame of video and then draw
+it on a `<canvas>`:
+
+    imageCapture.grabFrame()
+      .then(imageBitmap => {
+        canvas.width = imageBitmap.width;
+        canvas.height = imageBitmap.height;
+        canvas.getContext('2d').drawImage(imageBitmap, 0, 0);
+      })
+      .catch(error => console.error('grabFrame() error:', error));
+
+You can try this out in the demo below — if you're using a browser that supports the Image Capture API:
+
+* The `<video>` displays a stream from a camera via `getUserMedia()`.
+
+* If you have more than one camera, you can select the video source.
+
+* Tap or click the **GRAB FRAME** button to draw a frame from the video on a
+`<canvas>`.
+
+* Tap or click the **TAKE PHOTO** button to call `takePhoto()` and display the
+result in an `<img>`.
+
+* If the camera you are using supports zoom, a zoom slider will be displayed.
+
+Several variables from the demo are available from the console:
+
+* **constraints**
+* **imageCapture**
+* **mediaStream**
+
+<div id="ic-demo">
+  <button id="grabFrame">Grab Frame</button>
+  <button id="takePhoto">Take Photo</button>
+  <div class="select">
+    <label for="videoSource">Video source: </label><select id="videoSource"></select>
+  </div>
+  <input class="hidden" id="zoom" type="range" step="20">
+  <video autoplay class="hidden"></video>
+  <canvas class="hidden"></canvas>
+  <img class="hidden">
+</div>
+
+## Camera Capabilities
+
+In the demo above, you'll notice a difference in dimensions between the
+`grabFrame()` and `takePhoto()` results.
+
+The `takePhoto()` method gives access to the camera's maximum resolution.
+
+`grabFrame()` just takes the next-available `VideoFrame` in the `MediaStream`
+inside the renderer process, whereas `takePhoto()` interrupts the `MediaStream`,
+reconfigures the camera, takes the photo (usually in a compressed format,
+hence the `Blob`) and then resumes the `MediaStream`. In essence, this means
```

Comment:
Does this or can this cause any hiccup in the recorded video? You should probably answer that question either way.